### PR TITLE
Tests for many linalg methods

### PR DESCRIPTION
**Pull request by kshyatt**
Also fixed a bug in `diagonal.jl`.

### CODE REVIEW

**Comment by kshyatt** [comment]
cc @andreasnoack

File: base/linalg/diagonal.jl [hunk 1]
```diff
@@ -168,9 +168,9 @@ end
 
 A_mul_Bt(D::Diagonal, B::AbstractTriangular) = A_mul_B!(D, transpose(B))
 function A_mul_Bt(D::Diagonal, A::AbstractMatrix)
-    Ac = similar(A, promote_op(*, eltype(A), eltype(D.diag)), (size(A, 2), size(A, 1)))
-    ctranspose!(Ac, A)
-    A_mul_B!(D, Ac)
+    At = similar(A, promote_op(*, eltype(A), eltype(D.diag)), (size(A, 2), size(A, 1)))
```
Comment (tkelman) [inline base/linalg/diagonal.jl]: while you're adjusting this, I think it make more sense if the name of the second input were `B` rather than `A`

Comment (kshyatt) [inline base/linalg/diagonal.jl]: Do you want me to change the other methods too?

File: test/linalg/cholesky.jl [hunk 2]
```diff
@@ -69,6 +76,14 @@ for eltya in (Float32, Float64, Complex64, Complex128, BigFloat, Int)
     @test_approx_eq l*l' apd
     @test triu(capd.factors) ≈ lapd[:U]
     @test tril(lapd.factors) ≈ capd[:L]
+    if eltya <: Real
+        capds = cholfact(apds)
+        lapds = cholfact(apds, :L)
+        ls = lapds[:L]
+        @test_approx_eq ls*ls' apd
+        @test triu(capds.factors) ≈ lapds[:U]
+        @test tril(lapds.factors) ≈ capds[:L]
```
Comment (tkelman) [inline test/linalg/cholesky.jl]: how close do these end up being?

Comment (kshyatt) [inline test/linalg/cholesky.jl]: Unsure, I wanted to match the generic case here.

**Comment by kshyatt** [comment]
I'm going to merge, and come back later (or anyone else can feel free as well) to fix the variable names.

**Pull request merged**